foodnet/membership/auth_backends.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from django.contrib.auth import get_user_model
+from django.contrib.auth.backends import ModelBackend
+from allauth.account.models import EmailConfirmation
+
+from foodnet.membership.models import UserProfile
+
+
+log = logging.getLogger(__name__)
+
+
+class InvitationBackend(ModelBackend):
+
+    def authenticate(self, **credentials):
+        User = get_user_model()
+        email = credentials.get('email', credentials.get('username'))
+        verification_key = credentials.get('password')
+
+        if email:
+            users = User.objects.filter(email__iexact=email,
+                                        emailaddress__email__iexact=email,
+                                        emailaddress__verified=True)
+            confirmations = EmailConfirmation.objects.filter(
+                key=verification_key, email_address__verified=True)
+            if len(users) == 1 and len(confirmations) == 1:
```

Comment:
For efficiency `.count()` is preferred over `len()` if the only task is to count the number of rows. From Django docs:
```A count() call performs a SELECT COUNT(*) behind the scenes, so you should always use count() rather than loading all of the record into Python objects and calling len() on the result (unless you need to load the objects into memory anyway, in which case len() will be faster).```
https://docs.djangoproject.com/en/1.8/ref/models/querysets/#count

`len()` makes sense if you are using the result of the query though :)